View template or subplan list

JackSekoranja opened the pull request:
Completed Issue #37 (View Template or Subplan List (5)). 

This page is located at the address `http://127.0.0.1:8000/list`, and displays a list of templates and a list of subplans that are contained in the database. 

If you are notified that no templates/subplans exist, you need to add them (`http://127.0.0.1:8000/api/model/subplan/`, `http://127.0.0.1:8000/api/model/degree/`).

![Template:Subplan List](https://user-images.githubusercontent.com/36946090/55932122-a0c48300-5c6b-11e9-8ec5-e26403b65326.jpg)

Commit:
Added a courses table to the schema of models. Currently requisites and incompatibilities are unsupported.

Commit:
Added a subplan schema. Cleaned and reformatted schema implementation. Added fields to course model.

JackSekoranja committed:
Merge branch 'base-template' of https://github.com/cass-degrees/CASS-Degrees-Code into View_Template_or_Subplan_List

JackSekoranja committed:
Add a new page for JSON/database rendering

This commit adds a new page under `/list` that currently renders a predefined
JSON object, which will later be replaced with an API call to get live data

JackSekoranja committed:
Modified the list page to use Django template language rather than javascript

JackSekoranja committed:
Fix merge issues

Merge branches 'View_Template_or_Subplan_List' and 'develop' of https://github.com/cass-degrees/CASS-Degrees-Code into View_Template_or_Subplan_List

# Conflicts:
#	cassdegrees/api/admin.py
#	cassdegrees/api/models.py
#	cassdegrees/api/serializers.py
#	cassdegrees/api/urls.py
#	cassdegrees/api/views.py
#	cassdegrees/ui/urls.py
#	cassdegrees/ui/views.py

JackSekoranja committed:
Modified the list page to display actual template and subplan data from the database

JackSekoranja committed:
Merge branch 'develop' of https://github.com/cass-degrees/CASS-Degrees-Code into View_Template_or_Subplan_List

JackSekoranja committed:
Linked the new splash screen to the template/subplan list

j-selby commented on cassdegrees/templates/list.html:
```diff
@@ -0,0 +1,42 @@
+        {% endfor %}
+    </table>
+    {% if not subplans %}<p>No subplans to display</p>{% endif %}
+{% endblock %}
```
Newline.

j-selby commented on cassdegrees/templates/list.html:
```diff
@@ -0,0 +1,42 @@
+{% block subtitle %}Templates{% endblock %}
+
+{% block content %}
+    <table class="fullwidth tbl-cell-bdr">
```
Perhaps http://webpublishing.anu.edu.au/web-style-guide/page-tabs.php would be the way to go here.

j-selby commented on cassdegrees/ui/urls.py:
```diff
@@ -15,9 +15,10 @@
 urlpatterns = [
     path('', index),
     path('sampleform/', sampleform),
+    path('list', planList),
```
for consistency, should this end with `/`?

Daniel-Jang replied:
Yes it should, +1 here

j-selby commented on cassdegrees/ui/views.py:
```diff
@@ -14,12 +14,20 @@ def index(request):
         {'url': "/api/model/subplan/", 'img': "../static/img/create_subplan_img.png", 'label': "Create Subplan"},
         {'url': "", 'img': "../static/img/create_list_img.png", 'label': "Create List"},
-        {'url': "", 'img': "../static/img/open_existing_img.png", 'label': "Open Existing"},
+        {'url': "/list", 'img': "../static/img/open_existing_img.png", 'label': "Open Existing"},
```
Ditto with above.

j-selby commented on cassdegrees/ui/views.py:
```diff
@@ -14,12 +14,20 @@ def index(request):
 
+def planList(request):
+    # Generates a table based on whatever JSON object is stored in 'data'
+    # NOTE: data must be an array of dictionaries, with matching labels for each item
```
This should be a doc comment, no? I believe the plan was to use PEP-8, which would describe that.

j-selby reviewed (changes requested):
Looking good!

Daniel-Jang commented:
I'd be happy for you to pull to develop once those minor changes suggested by James has been addressed.

JackSekoranja committed:
Implemented suggested changes

• Added Newline to end of list.html
• Added `/` to the end of each url listing of `list`
• Added tabs to view the Templates/Subplans
• Used docstring instead of general comment

j-selby commented on cassdegrees/ui/views.py:
```diff
@@ -14,12 +14,22 @@ def index(request):
+    """
+    subplans = requests.get(request.build_absolute_uri('/api/model/subplan/?format=json')).json()
+    degree = requests.get(request.build_absolute_uri('/api/model/degree/?format=json')).json()
+    print("TEST", type(render(request, 'list.html', context={'subplans': subplans, 'degrees': degree})))
```
Is this TEST here for a reason?

JackSekoranja committed:
Removed debug code

pep8speaks commented:
Hello @JackSekoranja! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`cassdegrees/ui/views.py`](https://github.com/cass-degrees/CASS-Degrees-Code/blob/d1f77328ea33246f7ad6afa0684babe9ad428809/cassdegrees/ui/views.py):

> [Line 17:80](https://github.com/cass-degrees/CASS-Degrees-Code/blob/d1f77328ea33246f7ad6afa0684babe9ad428809/cassdegrees/ui/views.py#L17): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> [Line 23:1](https://github.com/cass-degrees/CASS-Degrees-Code/blob/d1f77328ea33246f7ad6afa0684babe9ad428809/cassdegrees/ui/views.py#L23): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 29:80](https://github.com/cass-degrees/CASS-Degrees-Code/blob/d1f77328ea33246f7ad6afa0684babe9ad428809/cassdegrees/ui/views.py#L29): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)
> [Line 30:80](https://github.com/cass-degrees/CASS-Degrees-Code/blob/d1f77328ea33246f7ad6afa0684babe9ad428809/cassdegrees/ui/views.py#L30): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 31:1](https://github.com/cass-degrees/CASS-Degrees-Code/blob/d1f77328ea33246f7ad6afa0684babe9ad428809/cassdegrees/ui/views.py#L31): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 32:80](https://github.com/cass-degrees/CASS-Degrees-Code/blob/d1f77328ea33246f7ad6afa0684babe9ad428809/cassdegrees/ui/views.py#L32): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
